### PR TITLE
Add snapshot pull command

### PR DESCRIPTION
**Pull request by abhijitrakas**

### CODE REVIEW

**Comment by abhijitrakas** [comment]
@thrijith Please review this PR when you get time.

**Review by thrijith — CHANGES_REQUESTED** [review]
Minor query/changes.

File: src/SnapshotCommand.php [hunk 1]
```diff
@@ -984,15 +982,224 @@ private function create_wp_content_backup() {
...
+
+	/**
+	 * Function to get file snapshot config file data.
+	 *
+	 * @param string $snapshot_name Snapshot name.
+	 *
+	 * @return mixed
+	 */
+	private function get_snapshot_file_data( $snapshot_name ) {
+		$zip_content   = $this->get_zip_contents( $snapshot_name ); // Get all the backup zip content.
```
Comment (thrijith) [inline src/SnapshotCommand.php:1167]:
```suggestion
		$zip_content = $this->get_zip_contents( $snapshot_name ); // Get all the backup zip content.
```

File: src/SnapshotStorage.php [hunk 2]
```diff
@@ -101,5 +101,36 @@ public function push_to_s3_bucket( $backup_info ) {
...
+							if ( ! empty( $upload_total_size ) ) {
+								$progress_percentage = number_format( ( $upload_size_so_far / $upload_total_size ) * 100, 2 ) . '%';
+								\WP_CLI::log( "Upload Progress : {$progress_percentage}" . PHP_EOL );
+							}
+						},
+					],
+				]
+			);
+			return true;
+		} catch ( S3Exception $e ) {
```
Comment (thrijith) [inline src/SnapshotStorage.php:131]:
```suggestion
		} catch ( \S3Exception $e ) {
```

**Review by thrijith — APPROVED** [review]

**Comment by abhijitrakas** [comment]
@thrijith - Please review it again. But don't merge.

**Review by thrijith — APPROVED** [review]

**Pull request merged**